academy/3-ibc/4-light-client-dev.md

```diff
@@ -0,0 +1,177 @@
+
+* [`VerifyPacketCommitment`](https://github.com/cosmos/ibc-go/blob/v7.0.0/modules/core/03-connection/keeper/verify.go#L205): to check if the proof added to a `MsgRecvPacket` submitted to the destination, points to a valid packet commitment on the source.
+* [`VerifyPacketAcknowledgement`](https://github.com/cosmos/ibc-go/blob/v7.0.0/modules/core/03-connection/keeper/verify.go#L250): to check if the proof added to a `MsgAcknowledgePacket` submitted to the source, points to a valid packet receipt commitment on the destination.
+* [`VerifyPacketReceiptAbsence`](https://github.com/cosmos/ibc-go/blob/v7.0.0/modules/core/03-connection/keeper/verify.go#L296): to check if the proof added to a `MsgTimeout` submitted to the source, proves that a packet receipt is absent at a height beyond the timeout height on the destination.
+
+All of the above rely on either `VerifyMembership` or `VerifyNonMembership` methods to prove either inclusion (also referred to as _existence_) or non-inclusion (_non-existence_) at a given commitment path.
+
+It is up to the light client developer to add these methods to their light client implementation.
+
+<HighlightBox type="reading">
+
+Please refer to the [ICS-23 implementation](https://github.com/cosmos/ibc-go/blob/v7.0.0/modules/core/23-commitment/types/merkle.go#L131-L205) for a concrete example.
+
+</HighlightBox>
+
+### Add light client to Cosmos SDK chain
+
+Now suppose that you have managed to develop a light client implementation fulfiling the requirements described above. How do you now get the light client on chain?
+
+This will depend on the chain environment you are in, so from here on out the assumption will be made that you will want to add the light client module to a Cosmos SDK chain.
+
+#### Configure light client module
+
+You will be adding your light client as an SDK module which must implement the SDK's [`AppModuleBasic`](https://github.com/cosmos/cosmos-sdk/blob/main/types/module/module.go#L50) interface.
+
+You must then register your light client module with `module.BasicManager` in the chain's `app.go` file.
+
+More information can be found [here](https://ibc.cosmos.network/main/ibc/light-clients/setup.html#configuring-a-light-client-module).
+
+#### Get support through governance
+
+In order to successfully create an IBC client using a new client type, it [must be supported](https://github.com/cosmos/ibc-go/blob/v7.0.0/modules/core/02-client/keeper/client.go#L19-L25). Light client support in IBC is gated by on-chain governance. The allow list may be updated by submitting a new governance proposal to update the `02-client` parameter `AllowedClients`.
+
+<HighlightBox type="info">
+
+To add your light client `0x-new-client` to `AllowedClients`, submit a proposal:
+
+```sh
+$ ... tx gov submit-proposal \
```

Comment:
Shouldn't this use the legacy command? We might need to update inc-go's documentation as well...

```suggestion
$ ... tx gov submit-legacy-proposal \
```

Reply:
@srdtrk Any input here?

Reply:
@xavierlepretre yes, it should be a legacy proposal since we haven't completed params migrations yet. I've done the transfer params migration only but it is not even merged to main yet. So all param chanegs should use legacy proposals